next permutation in c++

```diff
@@ -0,0 +1,69 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+//to print a vector
+void print(vector<int> a,int n){ 
+	for(int i=0;i<n;i++)
+		cout<<a[i]<<" ";
```

Comment:
This loop should print the vector. A better way to do this is to use a range based `for`-loop:
```c++
for (int value : a) {
  std::cout << value << " ";
}
```

Note that this - like your code - will always print a trailing whitespace, which may no be what you want.

Other notes:
1. You used post-increment, which is not idiomatic C++. While it does not matter in this case, getting into the habit of using pre-increment avoids unnecessary copies when dealing with iterators of containers with complex semantics. See https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement.
2. The `for`-loop body really should be in its own `{}`-block, even it if only has one statement. I find that this avoids many subtle bugs.

As a general comment, you should adopt a less compact formatting and apply it throughout. The Google style is a common one, see here: https://google.github.io/styleguide/cppguide.html#Formatting
You can also use an IDE and/or a command-line tool like clang-format to automate this.
Finally, https://github.com/google/styleguide has configuration files for various IDEs that you can use.